### PR TITLE
Simplify tests for job metrics by resetting them

### PR DESCRIPTION
**Pull request by alculquicondor**
#### What type of PR is this?

/kind cleanup

#### What this PR does / why we need it:

By resetting the metrics for each test, we don't need to query the old values.

#### Which issue(s) this PR fixes:

Fixes #

#### Does this PR introduce a user-facing change?

```release-note
NONE
```

#### Additional documentation e.g., KEPs (Kubernetes Enhancement Proposals), usage docs, etc.:

```docs

```

### CODE REVIEW

File: test/integration/job/job_test.go [hunk 1]
```diff
@@ -125,17 +125,6 @@ func TestMetrics(t *testing.T) {
...
-			jobFinishedNumBefore, err := getCounterMetricValueForLabels(metrics.JobFinishedNum, tc.wantJobFinishedNumMetricDelta.Labels)
-			if err != nil {
-				t.Fatalf("Failed to collect the JobFinishedNum metric before creating the job: %q", err)
-			}
-			jobPodsFinishedBefore, err := getCounterMetricValueForLabels(metrics.JobPodsFinished, tc.wantJobPodsFinishedMetricDelta.Labels)
-			if err != nil {
-				t.Fatalf("Failed to collect the JobPodsFinished metric before creating the job: %q", err)
-			}
-
 			// create a single job and wait for its completion
```
Comment (mimowo) [inline test/integration/job/job_test.go:128]: Call the `resetMetrics` function

**Review by mimowo — COMMENTED** [review]
lgtm, just call the function :)

File: test/integration/job/job_test.go (hunk 1 above)
Comment (alculquicondor) [reply test/integration/job/job_test.go:129]: oops, it was passing locally because the cases use different labels

**Comment by alculquicondor** [comment]
rebased

**Pull request merged**